python/shotgun_model/shotgun_query_model.py

```diff
@@ -170,10 +170,14 @@ def __init__(self, parent, bg_load_thumbs, bg_task_manager=None):
     ############################################################################
     # public methods
 
-    def clear(self):
+    def clear(self, remove_data_handler=True):
         """
         Removes all items (including header items) from the model and
         sets the number of rows and columns to zero.
+
+        :param bool remove_data_handler: Whether to remove the reference to the
```

Comment:
Curious why this is needed? Is it an optimization? It feels like we are exposing an internal implementation detail in the public interface - `data_handler` is an internal thing which I *think* is nothing we want to tell our users about (and may want to change in the future).

If it is an optimization, i think we should try to name it differently - we should also consider whether it's needed at all? The `hard_refresh` operation is a very low frequency operation.

Reply:
This has been refactored to remove the need for the new argument.